server/src/domain/person/person.service.ts

```diff
@@ -332,19 +332,26 @@ export class PersonService {
     this.logger.debug(`${faces.length} faces detected in ${asset.resizePath}`);
     this.logger.verbose(faces.map((face) => ({ ...face, embedding: `vector(${face.embedding.length})` })));
 
-    for (const face of faces) {
-      const mappedFace = {
-        assetId: asset.id,
-        embedding: face.embedding,
-        imageHeight: face.imageHeight,
-        imageWidth: face.imageWidth,
-        boundingBoxX1: face.boundingBox.x1,
-        boundingBoxX2: face.boundingBox.x2,
-        boundingBoxY1: face.boundingBox.y1,
-        boundingBoxY2: face.boundingBox.y2,
-      };
+    if (faces.length) {
+      await this.jobRepository.queue({ name: JobName.QUEUE_FACIAL_RECOGNITION, data: { force: false } });
+
+      const createFaces = faces.map((face) => {
+        const mappedFace = {
+          assetId: asset.id,
+          embedding: face.embedding,
+          imageHeight: face.imageHeight,
+          imageWidth: face.imageWidth,
+          boundingBoxX1: face.boundingBox.x1,
+          boundingBoxX2: face.boundingBox.x2,
+          boundingBoxY1: face.boundingBox.y1,
+          boundingBoxY2: face.boundingBox.y2,
+        };
+
+        return this.repository.createFace(mappedFace);
```

Comment:
Hmm, this is creating faces one at a time, but now you have changed it to run in parallel. It might be worth changing this to use something like `this.repository.createFaces` and pass in a list of faces instead. I know typeorm has some bulk insert methods (or save works in bulk as well).

Reply:
Oh right, good point.